trajan/waves/plot.py

```diff
@@ -69,8 +52,10 @@ def spectra(self, time, *args, **kwargs):
         """
         vrange = kwargs.pop('vrange', None)
         nseconds_gap = kwargs.pop('nseconds_gap', 6 * 3600)
-
-        ax = self.set_up_map(**kwargs)
+        try:
+            ax = kwargs.pop('ax')
+        except:
+            ax = plt.axes()
```

Comment:
This catches all exceptions. It is better to act as predetermined as possible, if there are exceptions there is probably an issue we should fix.

Reply:
Yes, they may be eagerly evaluated, maybe something like:

```
ax = kwargs.pop('ax', None)
ax = ax if ax is not None else plt.axes()
```

Reply:
I think your solution looks good.